src/incubator/panView/usePanGesture.ts

```diff
@@ -0,0 +1,141 @@
+import {useCallback} from 'react';
+import {PanGestureHandlerEventPayload} from 'react-native-gesture-handler';
+import {useSharedValue, withSpring, withTiming, useAnimatedGestureHandler, runOnJS} from 'react-native-reanimated';
+import {
+  PanningDirections,
+  PanningDirectionsEnum,
+  PanningDismissThreshold,
+  Frame,
+  getTranslation,
+  getDismissVelocity,
+  DEFAULT_THRESHOLD
+} from './panningUtil';
+import {HiddenLocation} from '../hooks/useHiddenLocation';
+
+export type PanViewDirections = PanningDirections;
+export const PanViewDirectionsEnum = PanningDirectionsEnum;
+export type PanViewDismissThreshold = PanningDismissThreshold;
+
+export interface PanGestureProps {
+  /**
+   * The directions of the allowed pan (default is all)
+   * Types: UP, DOWN, LEFT and RIGHT (using PanView.directions.###)
+   */
+  directions?: PanViewDirections[];
+  /**
+   * Dismiss the view if over the threshold (translation or velocity).
+   */
+  dismissible?: boolean;
+  /**
+   * Animate to start if not dismissed.
+   */
+  animateToOrigin?: boolean;
+  /**
+   * Callback to the dismiss animation end
+   */
+  onDismiss?: () => void;
+  /**
+   * Should the direction of dragging be locked once a drag has started.
+   */
+  directionLock?: boolean;
+  /**
+   * Object to adjust the dismiss threshold limits (eg {x, y, velocity}).
+   */
+  threshold?: PanViewDismissThreshold;
+  hiddenLocation: HiddenLocation;
```

Comment:
Does it make sense to call it `initialPosition`? 
We refer it as hidden because of the dialog but I think initial is more appropriate and clear.. WDYT?

Reply:
Actually it's a destination and not initial position; moreover, the logic is relying on it being of the type `HiddenLocation` (having `left`, `right`, `up` and `down`, which will be used to hide the view) so I'm not sure.
Thinking about it, perhaps the name of the component \ hook are wrong.